ceasiompy/CPACS2GMSH/func/generategmesh.py

```diff
@@ -0,0 +1,534 @@
+        ...
+        """
+        for bb in self.points_bb:
+            points_found = gmsh.model.occ.getEntitiesInBoundingBox(*bb, dim=0)
+            [self.points.append(point) for point in points_found]
+        self.points = set(self.points)
+
+        for bb in self.lines_bb:
+            lines_found = gmsh.model.occ.getEntitiesInBoundingBox(*bb, dim=1)
+            [self.lines.append(line) for line in lines_found]
+        self.lines = set(self.lines)
+
+        for bb in self.surfaces_bb:
+            surfaces_found = gmsh.model.occ.getEntitiesInBoundingBox(*bb, dim=2)
+            [self.surfaces.append(surface) for surface in surfaces_found]
+        self.surfaces = set(self.surfaces)
+
+
+# ==============================================================================
+#   FUNCTIONS
+# ==============================================================================
+
+
+def get_visual_bounding_box(dim_tag, enhance=0):
+    """
+    Function to find the bounding box (bb) of a gmsh object
+    The bounding box is extend by a fraction of its size in order to prevent
+    non detection of entites on the boundary of the original bb
+    ...
+
+    Args:
+    ----------
+    dim_tag : tuple
+        tuple (dim,tag) is the dim and tag of the gmsh volume of the airplane part
+    enhance : float
+        scaling factor to enhance the size of the bounding box
+    ...
+
+    Return :
+    ----------
+    bb : tuple
+        tuple containing the bottom right and top left location of the bb
+    size : tuple
+        tuple  containing the size of the bb in each dimention
+    """
+
+    bb = list(gmsh.model.occ.getBoundingBox(*dim_tag))
+    sizes = [abs(bb[0] - bb[3]), abs(bb[1] - bb[4]), abs(bb[2] - bb[5])]
+
+    # Extend a bit the bounding_box if enhance is non zero
+    bb[0] = bb[0] - sizes[0] * enhance
+    bb[1] = bb[1] - sizes[1] * enhance
+    bb[2] = bb[2] - sizes[2] * enhance
+    bb[3] = bb[3] + sizes[0] * enhance
+    bb[4] = bb[4] + sizes[1] * enhance
+    bb[5] = bb[5] + sizes[2] * enhance
+
+    # Update size due to the extend
+    sizes = (abs(bb[0] - bb[3]), abs(bb[1] - bb[4]), abs(bb[2] - bb[5]))
+    return (bb[0], bb[1], bb[2], bb[3], bb[4], bb[5]), sizes
+
+
+def hierarchy_surface(part1, part2, surfaces, operation):
+    """
+    Function to select which of two part sharing an common surface should be the final
+    parent of the surface
+    ...
+
+    Args:
+    ----------
+    part1 : AircraftPart
+        aircraft part sharing a surface with part2
+    part2 : AircraftPart
+        aircraft part sharing a surface with part1
+    operation : int
+        operation to perform on the the surface : 0 := remove, 1 := add
+    ...
+    """
+
+    # Score : a part with a higher score will be given priority to the surface assignation
+    def give_score(part):
+        if "fuselage" in part.name:
+            return 2
+        if "wing" in part.name:
+            return 1
+        if "engine" in part.name:
+            return 3
```

Comment:
It could be a property of the AircraftPart Class no?

And it will probably necessary to have more flexibility for the names, e.g. ["Fuselage", "fuslage", "Fus", "fus", ..]

Reply:
These names are defined (by us) when we export the .brep file with tigl, so they are always the same and it is intended that we can identify each part of the plane clearly: fuselage1, fuselage2, wing1, wing1_m, ...

For the scores, of course, this could be a class property, I will move this piece of code where the aircraft parts are defined.